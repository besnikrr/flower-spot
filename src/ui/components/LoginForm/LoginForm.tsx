import { TextField } from '@mui/material';
import { useContext } from 'react';
import { useForm } from 'react-hook-form';
import { AuthContext } from '../../../application/authContext';
import { UserLogin, userLoginValidator } from '../../../domain/user';
import { login } from '../../../services/user';
import { yupResolver } from '@hookform/resolvers/yup';

interface LoginFormProps {
  onClose?: (value: boolean) => void;
}

export const LoginForm = ({ onClose }: LoginFormProps): JSX.Element => {
  const authContext = useContext(AuthContext);

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<UserLogin>({
    resolver: yupResolver(userLoginValidator),
  });

  const onSubmit = async (data: UserLogin) => {
    const result = await login(data);
    authContext.login(result.data.auth_token);
    onClose?.(true);
  };

  return (
    <form className="modal" onSubmit={handleSubmit(onSubmit)}>
      <h1>Welcome Back</h1>
      <TextField
        className="text-field"
        label="Email Address"
        type="email"
        InputLabelProps={{
          shrink: true,
        }}
        variant="filled"
        {...register('email')}
        error={!!errors?.email}
      />
      <TextField
        className="text-field"
        label="Password"
        type="password"
        InputLabelProps={{
          shrink: true,
        }}
        variant="filled"
        {...register('password')}
        error={!!errors?.password}
      />
      <button
        type="submit"
        className={'btn btn--pink btn--squared full-width create-btn'}
      >
        Login to your Account
      </button>
    </form>
  );
};
