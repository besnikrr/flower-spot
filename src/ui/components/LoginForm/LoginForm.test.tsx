import { LoginForm } from './LoginForm';
import renderer from 'react-test-renderer';

// SNAPSHOT Test
it('renders correctly', () => {
  const tree = renderer.create(<LoginForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
