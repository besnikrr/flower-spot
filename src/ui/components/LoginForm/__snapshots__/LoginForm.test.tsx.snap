// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="modal"
  onSubmit={[Function]}
>
  <h1>
    Welcome Back
  </h1>
  <div
    className="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Email Address
    </label>
    <div
      className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id=":r0:"
        name="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="email"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r1:"
      id=":r1:-label"
    >
      Password
    </label>
    <div
      className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id=":r1:"
        name="password"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
      />
    </div>
  </div>
  <button
    className="btn btn--pink btn--squared full-width create-btn"
    type="submit"
  >
    Login to your Account
  </button>
</form>
`;
